controllers/specialresourcepreflight_controller.go

```diff
@@ -0,0 +1,214 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/go-logr/logr"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+
+	srov1beta1 "github.com/openshift/special-resource-operator/api/v1beta1"
+	"github.com/openshift/special-resource-operator/internal/controllers/state"
+	"github.com/openshift/special-resource-operator/pkg/clients"
+	"github.com/openshift/special-resource-operator/pkg/cluster"
+	"github.com/openshift/special-resource-operator/pkg/helmer"
+	"github.com/openshift/special-resource-operator/pkg/preflight"
+	"github.com/openshift/special-resource-operator/pkg/runtime"
+	"github.com/openshift/special-resource-operator/pkg/utils"
+)
+
+// ClusterPreflightReconciler reconciles a SpecialResourcePreflight object
+type SpecialResourcePreflightReconciler struct {
+	Log           logr.Logger
+	ClusterAPI    cluster.Cluster
+	Helmer        helmer.Helmer
+	PreflightAPI  preflight.PreflightAPI
+	RuntimeAPI    runtime.RuntimeAPI
+	StatusUpdater state.StatusUpdater
+	KubeClient    clients.ClientsInterface
+}
+
+func (r *SpecialResourcePreflightReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		For(&srov1beta1.SpecialResourcePreflight{}).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: 1,
+		}).
+		Complete(r)
+}
+
+// Reconcile Reconiliation entry point
+func (r *SpecialResourcePreflightReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+
+	reconRes := ctrl.Result{}
+
+	log := r.Log.WithName(utils.Print("clusterpreflight", utils.Brown))
```

Comment:
Seems like we have two loggers here (that and `r.Log`). Is this a temporary until we have a logging strategy?

Reply:
I think there's no need to keep a `Log` field in the controller struct. The manager already injects one in `ctx`, so this should work:
```suggestion
	log := logger := log.FromContext(ctx)
```

Reply:
Lets leave all this changes for the logging epic